htlcswitch/link.go

```diff
@@ -760,6 +818,26 @@ func (l *channelLink) syncChanStates() error {
 			}
 		}
 
+		// If we have been signaled to retransmit Shutdown, we'll do so
+		// here.
+		if l.cfg.RetransmitShutdown {
+			// Recreate the Shutdown message.
+			shutdownMsg := lnwire.NewShutdown(
+				l.ChanID(),
+				l.channel.LocalUpfrontShutdownScript(),
+			)
+
+			// Flip the shutdownInit and shutdownSent bools.
+			l.shutdownInit = true
+			l.shutdownSent = true
```

Comment:
I think we need to set `l.shutdownSent` after sending the message below. Suppose anything goes wrong with `l.cfg.Peer.SendMessage`, we actually didn't send the msg.

Reply:
if there is an error returned from `SendMessage`, then we just exit with an error here and close up the link. we should then retransmit the next time the link is started